tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,244 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{
+        extensions::{Extensions, ExtensionsInner, ExtensionsMut},
+        LookupMetadata, LookupSpan, SpanData,
+    },
+    sync::RwLock,
+};
+use std::{
+    cell::RefCell,
+    sync::atomic::{fence, AtomicUsize, Ordering},
+};
+use tracing_core::{
+    dispatcher,
+    field::FieldSet,
+    span::{self, Current, Id},
+    Event, Interest, Metadata, Subscriber,
+};
+
+/// `Registry` is a shared, reusable store for spans.
```

Comment:
Nit: usually rustdoc comments _don't_ begin with the name of the type, this should maybe be 
```suggestion
/// A shared, reusable store for spans.
```

Reply:
since this is specifically a sharded registry (it lives in the `Sharded` module) maybe we should note that it uses a sharded slab for storage?